src/Catalyst.Core.Modules.Dfs/UnixFileSystem/FileSystem.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs.CoreApi;
+using Catalyst.Core.Modules.Keystore;
+using Lib.P2P;
+using ProtoBuf;
+
+namespace Catalyst.Core.Modules.Dfs.UnixFileSystem
+{
+    /// <summary>
+    ///   Support for the *nix file system.
+    /// </summary>
+    public static class FileSystem
+    {
+        static readonly byte[] emptyData = new byte[0];
+
+        /// <summary>
+        ///   Creates a stream that can read the supplied <see cref="Cid"/>.
+        /// </summary>
+        /// <param name="id">
+        ///   The identifier of some content.
+        /// </param>
+        /// <param name="blockService">
+        ///   The source of the cid's data.
+        /// </param>
+        /// <param name="keyChain">
+        ///   Used to decypt the protected data blocks.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation. The task's value is
+        ///   a <see cref="Stream"/> that produces the content of the <paramref name="id"/>.
+        /// </returns>
+        /// <remarks>
+        ///  The id's <see cref="Cid.ContentType"/> is used to determine how to read
+        ///  the conent.
+        /// </remarks>
+        public static Task<Stream> CreateReadStreamAsync(Cid id,
+            IBlockApi blockService,
+            KeyChain keyChain,
+            CancellationToken cancel)
+        {
+            // TODO: A content-type registry should be used.
+            if (id.ContentType == "dag-pb")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)